pkg/sql/compile/scopeRemoteRun.go

```diff
@@ -116,13 +117,18 @@ func CnServerMessageHandler(
 	udfService udf.Service,
 	cli client.TxnClient,
 	aicm *defines.AutoIncrCacheManager,
-	messageAcquirer func() morpc.Message) error {
+	messageAcquirer func() morpc.Message) (err error) {
 	defer func() {
 		if e := recover(); e != nil {
-			err := moerr.ConvertPanicError(ctx, e)
+			err = moerr.ConvertPanicError(ctx, e)
 			getLogger().Error("panic in cn message handler",
 				zap.String("error", err.Error()))
-			cs.Close()
+
+			err2 := cs.Close()
+			if err2 != nil {
+				err = errors.Join(err, err2)
+			}
```

Comment:
errors.Join will discard nil errors, so err = errors.Join(err, cs.Close()) will be OK

Reply:
Thank you for this feedback. Updated accordingly.